ctx.go

```diff
@@ -365,6 +365,41 @@ func (c *netHttpContext[B, P]) Body() (B, error) {
 	return body, err
 }
 
+// setParamValue sets a value to a reflect.Value based on its kind
+func setParamValue(value reflect.Value, paramValue string, kind reflect.Kind) error {
+	switch kind {
+	case reflect.String:
+		value.SetString(paramValue)
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		intValue, err := strconv.Atoi(paramValue)
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to int: %w", paramValue, err)
+		}
+		value.SetInt(int64(intValue))
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		uintValue, err := strconv.ParseUint(paramValue, 10, 64)
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to uint: %w", paramValue, err)
+		}
+		value.SetUint(uintValue)
+	case reflect.Float32, reflect.Float64:
+		floatValue, err := strconv.ParseFloat(paramValue, 64)
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to float64: %w", paramValue, err)
+		}
+		value.SetFloat(floatValue)
+	case reflect.Bool:
+		boolValue, err := strconv.ParseBool(paramValue)
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to bool: %w", paramValue, err)
+		}
+		value.SetBool(boolValue)
+	default:
+		return fmt.Errorf("unsupported type %s", kind)
+	}
+	return nil
+}
+
 func (c *netHttpContext[B, P]) Params() (P, error) {
 	p := new(P)
```

Comment:
Nit: This is old code, but why creating a pointer, when method return the non-pointer version